lib/elixir/lib/kernel.ex

```diff
@@ -4117,6 +4117,64 @@ defmodule Kernel do
   end
 
   @doc """
+  Generates a macro suitable for use in guard expressions.
+
+  It raises at compile time if the definition uses expressions that aren't
+  allowed in guards, and otherwise creates a macro that can be used both inside
+  or outside guards.
+
+  ## Example
+
+      defmodule Integer.Guards do
+        defguard is_even(value) when is_integer(value) and rem(value, 2) == 0
+      end
+
+      defmodule Integer.Utils do
+        import Integer.Guards
+
+        def is_even_and_large?(value) when is_even(value) and value > 100, do: true
+        def is_even_and_large?(_), do: false
+
+        def is_even_and_small?(value) do
+          if is_even(value) and value <= 100, do: true, else: false
+        end
```

Comment:
Having 2 separate examples would work just as well. You could add a paragraph saying

> Just like the builtin the guards, the ones defined with `defguard` can also be used like normal functions:
>
> ```elixir
> if is_even(num), do: <something>
> ```